### PR TITLE
fix(setup): ignore comment lines in env files

### PR DESCRIPTION
**Pull request by kenji-miyake**
## Description

Split from autowarefoundation/autoware#248.
Same as https://github.com/autowarefoundation/autoware/blob/14477196ae5435895f55b7c42cb9956b33d6a6eb/.github/workflows/docker-build-and-push.yaml#L23.

## Pre-review checklist for the PR author

The PR author **must** check the checkboxes below when creating the PR.

- [x] I've confirmed the [contribution guidelines].
- [x] The PR follows the [pull request guidelines].

## In-review checklist for the PR reviewers

The PR reviewers **must** check the checkboxes below before approval.

- [x] The PR follows the [pull request guidelines].

## Post-review checklist for the PR author

The PR author **must** check the checkboxes below before merging.

- [x] There are no open discussions or they are tracked via tickets.

After all checkboxes are checked, anyone who has write access can merge the PR.

[contribution guidelines]: https://autowarefoundation.github.io/autoware-documentation/main/contributing/
[pull request guidelines]: https://autowarefoundation.github.io/autoware-documentation/main/contributing/pull-request-guidelines/

### CODE REVIEW

File: setup-dev-env.sh [hunk 1]
```diff
@@ -73,7 +73,7 @@ fi
 
 # Add env args
 # shellcheck disable=SC2013
-for env_name in $(sed "s/=.*//" <amd64.env); do
+for env_name in $(sed -e "/^#/d" -e "s/=.*//" <amd64.env); do
```
Comment (xmfcx) [inline setup-dev-env.sh]: I've checked `sed -e "/^#/d" -e "s/=.*//" <(echo "// test //aaa bbb")` on terminal and it outputs the given string. 

What is this `sed` command supposed to do?

Comment (kenji-miyake) [reply setup-dev-env.sh]: I'm sorry for the lack of my explanation. :bow: 

It just removes `# Comment` lines.
Could you test with this env file?

```
# comment1
rosdistro=humble
rmw_implementation=rmw_cyclonedds_cpp
cuda_version=11.6
# comment2
cuda_image_tag=11.6.2-devel-ubuntu20.04
cudnn_version=8.4.0.27-1+cuda11.6
tensorrt_version=8.2.4-1+cuda11.4
```

Comment (kenji-miyake) [reply setup-dev-env.sh]: Without this PR (`--extra-vars #=0 --extra-vars comment2=` is wrong):

```
❯ ./setup-dev-env.sh
Setting up the build environment can take up to 1 hour.
>  Are you sure you want to run setup? [y/N] y
Starting galaxy collection install process
Process install dependency map
Starting collection install process
Installing 'autoware.dev_env:0.1.0' to '/home/kenji/.ansible/collections/ansible_collections/autoware/dev_env'
Created collection for autoware.dev_env:0.1.0 at /home/kenji/.ansible/collections/ansible_collections/autoware/dev_env
autoware.dev_env:0.1.0 was installed successfully
 Run ansible-playbook autoware.dev_env.universe --ask-become-pass --extra-vars #=0 --extra-vars comment1= --extra-vars rosdistro=humble --extra-vars rmw_implementation=rmw_cyclonedds_cpp --extra-vars cuda_version=11.6 --extra-vars #=0 --extra-vars comment2= --extra-vars cuda_image_tag=11.6.2-devel-ubuntu20.04 --extra-vars cudnn_version=8.4.0.27-1+cuda11.6 --extra-vars tensorrt_version=8.2.4-1+cuda11.4 
```

With this PR:

```
❯ ./setup-dev-env.sh
Setting up the build environment can take up to 1 hour.
>  Are you sure you want to run setup? [y/N] y
Starting galaxy collection install process
Process install dependency map
Starting collection install process
Installing 'autoware.dev_env:0.1.0' to '/home/kenji/.ansible/collections/ansible_collections/autoware/dev_env'
Created collection for autoware.dev_env:0.1.0 at /home/kenji/.ansible/collections/ansible_collections/autoware/dev_env
autoware.dev_env:0.1.0 was installed successfully
 Run ansible-playbook autoware.dev_env.universe --ask-become-pass --extra-vars rosdistro=humble --extra-vars rmw_implementation=rmw_cyclonedds_cpp --extra-vars cuda_version=11.6 --extra-vars cuda_image_tag=11.6.2-devel-ubuntu20.04 --extra-vars cudnn_version=8.4.0.27-1+cuda11.6 --extra-vars tensorrt_version=8.2.4-1+cuda11.4 
```

Comment (xmfcx) [reply setup-dev-env.sh]: Ok I was testing with `// style comments, not # `.

Yes I confirm it works with `# test` but it works but doesn't work if there are spaces before the `#`. 
Example: 
```
# removes this line
 # doesn't remove this line
```

Following will remove the whitespace before the line (ref: https://quickref.me/sed):
```suggestion
for env_name in $(sed -e "s/^\s*//" -e "/^#/d" -e "s/=.*//" <amd64.env); do
```

Comment (kenji-miyake) [reply setup-dev-env.sh]: Yeah, but I thought we can prevent the case by reviews.

If we update it, we also have to update this line.
https://github.com/autowarefoundation/autoware/blob/14477196ae5435895f55b7c42cb9956b33d6a6eb/.github/workflows/docker-build-and-push.yaml#L23

I'll update both.

Comment (kenji-miyake) [reply setup-dev-env.sh]: Fixed in aa550a32b1da3742e3f9ce55ef35689f5ce4b8c3.

Comment (xmfcx) [reply setup-dev-env.sh]: Thanks!

**Review by xmfcx — APPROVED** [review]

**Pull request merged**